fv/wireguard_test.go

```diff
@@ -875,9 +876,163 @@ var _ = infrastructure.DatastoreDescribe("_BPF-SAFE_ WireGuard-Supported 3 node
+
+			tcpdump.Start()
+			tcpdumps = append(tcpdumps, tcpdump)
+		}
+
+		cc = &connectivity.Checker{}
+	})
+
+	AfterEach(func() {
+		if CurrentGinkgoTestDescription().Failed {
+			for _, felix := range felixes {
+				felix.Exec("ip", "addr")
+				felix.Exec("ip", "rule", "list")
+				felix.Exec("ip", "route", "show", "table", "all")
+				felix.Exec("ip", "route", "show", "cached")
+				felix.Exec("wg")
+				felix.Exec("iptables-save", "-t", "raw")
+				felix.Exec("cat", "/proc/sys/net/ipv4/conf/all/src_valid_mark")
+			}
+		}
+
+		for _, wl := range wls {
+			wl.Stop()
+		}
+
+		for _, tcpdump := range tcpdumps {
+			tcpdump.Stop()
+		}
+
+		for _, felix := range felixes {
+			felix.Stop()
+		}
+
+		if CurrentGinkgoTestDescription().Failed {
+			infra.DumpErrorData()
+		}
+		infra.Stop()
+	})
+
+	It("should pass basic connectivity scenarios", func() {
+		// Check that felix-0, felix-1 is ready
+		// 1. by checking, Wireguard interface exist.
+		By("Checking the wireguard links")
+		Eventually(func() error {
+			for i := range []int{0, 1} {
+				out, err := felixes[i].ExecOutput("ip", "link")
+				if err != nil {
+					return err
+				}
+				if strings.Contains(out, wireguardInterfaceNameDefault) {
+					continue
+				}
+				return fmt.Errorf("felix-%d has no wireguard device", i)
+			}
+			return nil
+		}, "10s", "100ms").ShouldNot(HaveOccurred())
+		// 2. by checking, Wireguard rule exist.
+		for i := range []int{0, 1} {
+			Eventually(func() string {
+				return getWireguardRoutingRule(felixes[i])
+			}, "10s", "100ms").Should(MatchRegexp(fmt.Sprintf("\\d+:\\s+from all fwmark 0/0x\\d+ lookup \\d+")))
+		}
+		// 3. by checking, Wireguard route table exist.
+		Eventually(func() []string {
+			return strings.Split(getWireguardRouteEntry(felixes[0]), "\n")
+		}, "10s", "100ms").Should(ContainElements(
+			ContainSubstring(fmt.Sprintf("%s dev wireguard.cali scope link", felixes[1].IP)),
+			ContainSubstring(fmt.Sprintf("%s dev wireguard.cali scope link", wls[1].IP)),
+		))
+		Eventually(func() []string {
+			return strings.Split(getWireguardRouteEntry(felixes[1]), "\n")
+		}, "10s", "100ms").Should(ContainElements(
+			ContainSubstring(fmt.Sprintf("%s dev wireguard.cali scope link", felixes[0].IP)),
+			ContainSubstring(fmt.Sprintf("%s dev wireguard.cali scope link", wls[0].IP)),
+		))
+
+		By("Checking the proc/sys src valid mark entries")
+		for _, felix := range felixes {
+			Eventually(func() string {
+				s, _ := felix.ExecCombinedOutput("cat", "/proc/sys/net/ipv4/conf/all/src_valid_mark")
+				return s
+			}, "10s", "100ms").Should(ContainSubstring("1"))
+		}
+
+		cc.ExpectSome(wls[0], wls[1])
+		cc.ExpectSome(wls[1], wls[0])
+		cc.CheckConnectivityWithTimeout(30 * time.Second)
+
+		By("verifying packets between felix-0 and felix-1 is encrypted")
+		for i := range []int{0, 1} {
+			Eventually(func() int {
+				return tcpdumps[i].MatchCount("numTunnelPacketsFelix0toFelix1")
```

Comment:
Tip: we've got `tcpdump.MatchCountFn` for this case now; avoids the extra `func()`

Reply:
Thanks, updated the code to use this.